include/deal.II/sundials/sunlinsol_wrapper.h

```diff
@@ -168,6 +170,39 @@ namespace SUNDIALS
     public:
       explicit LinearSolverWrapper(LinearSolveFunction<VectorType> lsolve);
 
+      /**
+       * Wrap the given @p solver.
+       */
+      template <typename SolverType>
+      LinearSolverWrapper(SolverType &solver)
+        : LinearSolverWrapper([&](SundialsOperator<VectorType> &      op,
+                                  SundialsPreconditioner<VectorType> &prec,
+                                  VectorType &                        x,
+                                  const VectorType &                  b,
+                                  double                              tol) {
+          (void)tol;
+          solver.solve(op, x, b, prec);
+          return 0;
+        })
+      {}
```

Comment:
To get a working version I implement the new way in terms of the old std::function. In the future, I'd like to removed the std::function and instead only support the new way. Then, the wrapper can implement some more of the API specified by SUNDIALS, e.g. a function returning the number of iterations.